Lib/test/test_traceback.py

```diff
@@ -3802,6 +3802,26 @@ def test_traceback_header(self):
         exc = traceback.TracebackException(Exception, Exception("haven"), None)
         self.assertEqual(list(exc.format()), ["Exception: haven\n"])
 
+    def test_exception_punctuation_handling_with_suggestions(self):
+        def raise_mssage(message, name):
+            try:
+                raise NameError(message, name=name)
+            except Exception as e:
+                return traceback.TracebackException.from_exception(e)._str
+
+        test_cases = [
+            ("Error.", "time", "Error. Did you forget to import 'time'?"),
+            ("Error?", "time", "Error? Did you forget to import 'time'?"),
+            ("Error!", "time", "Error! Did you forget to import 'time'?"),
+            ("Error", "time", "Error. Did you forget to import 'time'?"),
```

Comment:
Please add a test where there is the "Or did you forget [...]" part.

Reply:
I would forget about the fact that it's in the IDLE module. We sometimes have multiple tests covered here and there. This PR refactors code, so it's still good to have coverage even if it's covered twice. Having it covered where it *should* be covered directly would be better. In IDLE, it's an indirect coverage as it doesn't explicitly tests traceback itself IMO.